api/handlers.go

```diff
@@ -95,6 +106,41 @@ func (api *API) unpinPOST(w http.ResponseWriter, req *http.Request, _ httprouter
 	api.WriteSuccess(w)
 }
 
+// serverRemovePOST informs pinner that a given server is dead and should be removed as
+// pinner from all skylinks it's marked as pinning.
+func (api *API) serverRemovePOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var body ServerRemoveRequest
+	err := json.NewDecoder(req.Body).Decode(&body)
+	if err != nil {
+		api.WriteError(w, err, http.StatusBadRequest)
+		return
+	}
+	if body.Server == "" {
+		api.WriteError(w, errors.New("no server found in request body"), http.StatusBadRequest)
+		return
+	}
+	// Remove the server as pinner.
+	n, err := api.staticDB.RemoveServer(req.Context(), body.Server)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "failed to remove server"), http.StatusInternalServerError)
+		return
+	}
+	// Schedule a scan for underpinned skylinks in an hour, so all of them can
```

Comment:
`nit` maybe move this up before the server removal? I think that's safer because if scheduling fails the server won't be removed, and if the server's removed but scheduling fails, that's annoying.

Reply:
Sounds good. I'll do it in a F/U, so we don't need to re-approve. I'll keep this open until then, so I don't forget.

Reply:
Quickly hijacking this to ask a question. If a server is removed and we schedule a scan for 1 hour in the future. Then we remove another one 30 minutes later. Will the scan happen at the original time or 1 hour from the second removal?

Reply:
This is a good point. Servers check for the next scan each 30 minutes until the next scan is scheduled for less than 30 minutes in the future. When a server sees that a scan is scheduled in less than 30 minutes it sleeps its scanner until the exact moment of the scan and performs it without checking again. I did that in order to avoid endless checking and some complications caused by clock desynchronisation between servers.

So, if two servers are checking for next scan and server1 checks 35 minutes before the scan it will have to check again 5 minutes before the scan again. if server2 checks 25 minutes before the scan it will not check again. Then, if we delete a server 20 minutes before the scan then a new scan will be scheduled for one hour later. server1 will check and will wait for that hour before scanning. Server2 will not - it will sleep for its 25 minutes and then do a scan.

I believe the best approach here is to check for next scan before scheduling one and if it's in less than an hour just not schedule one because we've got it coming anyway.

Good catch! I'll implement it in #45 as F/U.